cli/internal/transformerpipeline/pipeline.go

```diff
@@ -82,6 +87,13 @@ func (lp *TransformerPipeline) OnOutput(fn func([]byte) error) error {
 }
 
 func (lp *TransformerPipeline) Close() error {
+	// Closing the pipeline happens on both source as well as destination close.
```

Comment:
I think we should change this function signature to `Close()`, that is, not return `error`.

I honestly don't understand what it means for `CloseSend()` to return an error, or what I'd do differently in either case, but we're not gonna do anything, because once `closed` is `true` we won't call it again.

This also simplifies `sync_v3`, because we don't have to check the `error`. I think it's wrong to `return err` where the pipeline is being closed anyway.

Reply:
Can also use `sync.Once`?

Reply:
Done